data/mods/gen5/conditions.ts

```diff
@@ -32,8 +32,7 @@ export const Conditions: import('../../../sim/dex-conditions').ModdedConditionDa
 			// However, just in case, use 1 if it is undefined.
 			const counter = this.effectState.counter || 1;
 			if (counter >= 256) {
-				// 2^32 - special-cased because Battle.random(n) can't handle n > 2^16 - 1
```

Comment:
Holy shit, I traced this back to

https://github.com/smogon/pokemon-showdown/commit/9bbb0b8d567375b548682d27f64831c465c264c2#diff-1876d2faa56ede4eb79712f3c52e03d209fbd5e4b99ad47f5d1b31d3197fbfafR8460

Back then we were using the Gen 3 RNG, which only generated 16-bit numbers. Very different times.